language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/FromClauseNodeContext.java

```diff
@@ -123,19 +122,31 @@ public void sort(BallerinaCompletionContext context,
                 TypeDescKind.STRING, TypeDescKind.ARRAY,
                 TypeDescKind.MAP, TypeDescKind.TABLE,
                 TypeDescKind.STREAM, TypeDescKind.XML);
-        
-        completionItems.forEach(lsCItem -> {
-            String sortText = SortingUtil.genSortText(3) + 
-                    SortingUtil.genSortText(SortingUtil.toRank(context, lsCItem));
-            if (CommonUtil.isCompletionItemOfType(lsCItem, iterables)) {
-                sortText = SortingUtil.genSortText(1)
-                        + SortingUtil.genSortText(SortingUtil.toRank(context, lsCItem));
-            } else if (lsCItem.getType() == LSCompletionItem.CompletionItemType.SYMBOL &&
-                    lsCItem.getCompletionItem().getKind() == CompletionItemKind.Function) {
-                sortText = SortingUtil.genSortText(2);
-            }
-            lsCItem.getCompletionItem().setSortText(sortText);
-        });
+
+        Optional<TypeSymbol> expectedTypeSymbol = context.getContextType();
+        if (context.getCursorPositionInTree() > node.inKeyword().position()) {
+            completionItems.forEach(lsCItem -> {
+                String sortText = SortingUtil.genSortText(3) +
+                        SortingUtil.genSortText(SortingUtil.toRank(context, lsCItem));
+                if (expectedTypeSymbol.isPresent()
+                        && SortingUtil.isCompletionItemAssignable(lsCItem, expectedTypeSymbol.get())) {
+                    sortText = SortingUtil.genSortText(1)
+                            + SortingUtil.genSortText(SortingUtil.toRank(context, lsCItem));
+                } else if (CommonUtil.isCompletionItemOfType(lsCItem, iterables)) {
+                    sortText = SortingUtil.genSortText(1)
+                            + SortingUtil.genSortText(SortingUtil.toRank(context, lsCItem, 2));
+                }
+                lsCItem.getCompletionItem().setSortText(sortText);
```

Comment:
We can simplify this as follows
```suggestion
                int rank =3
                if (expectedTypeSymbol.isPresent()
                        && SortingUtil.isCompletionItemAssignable(lsCItem, expectedTypeSymbol.get())) {
                    rank = 1
                } else if (CommonUtil.isCompletionItemOfType(lsCItem, iterables)) {
                    rank = 2
                }
                String sortText = SortingUtil.genSortText(rank) +
                        SortingUtil.genSortText(SortingUtil.toRank(context, lsCItem));
                lsCItem.getCompletionItem().setSortText(sortText);
```